src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/EndToEndTests.cs

```diff
@@ -431,6 +431,82 @@ public async Task EndToEnd_MultiProjectSolution()
+    /// enabled by default, so the target will be skipped.
+    /// </summary>
+    [DockerAvailableFact]
+    public async Task EndToEnd_MultiProjectSolution_with_multitargeted_library()
+    {
+        ILogger logger = _loggerFactory.CreateLogger(nameof(EndToEnd_MultiProjectSolution_with_multitargeted_library));
+        DirectoryInfo newSolutionDir = new(Path.Combine(TestSettings.TestArtifactsDirectory, nameof(EndToEnd_MultiProjectSolution_with_multitargeted_library)));
+
+        if (newSolutionDir.Exists)
+        {
+            newSolutionDir.Delete(recursive: true);
+        }
+
+        newSolutionDir.Create();
+
+        // Create solution with projects
+        new DotnetNewCommand(_testOutput, "sln", "-n", nameof(EndToEnd_MultiProjectSolution_with_multitargeted_library))
+            .WithVirtualHive()
+            .WithWorkingDirectory(newSolutionDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        new DotnetNewCommand(_testOutput, "web", "-n", "WebApp")
+            .WithVirtualHive()
+            .WithWorkingDirectory(newSolutionDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        new DotnetCommand(_testOutput, "sln", "add", Path.Combine("WebApp", "WebApp.csproj"))
+            .WithWorkingDirectory(newSolutionDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        new DotnetNewCommand(_testOutput, "classlib", "-n", "Library")
+            .WithVirtualHive()
+            .WithWorkingDirectory(newSolutionDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        new DotnetCommand(_testOutput, "sln", "add", Path.Combine("Library", "Library.csproj"))
+            .WithWorkingDirectory(newSolutionDir.FullName)
+            .Execute()
+            .Should().Pass();
+
+        // Set TFMs for Library - use current toolset + NS2.0 for compatibility
+        // also set IsPublishable to false
+        using (FileStream stream = File.Open(Path.Join(newSolutionDir.FullName, "Library", "Library.csproj"), FileMode.Open, FileAccess.ReadWrite))
+        {
+            XDocument document = await XDocument.LoadAsync(stream, LoadOptions.None, CancellationToken.None);
+            var tfmNode =
+                document
+                .Descendants()
+                .First(e => e.Name.LocalName == "TargetFramework");
+            var propertyGroupNode = tfmNode.Parent!;
+            tfmNode.Remove();
+            propertyGroupNode.Add(new XElement("TargetFrameworks", $"{ToolsetInfo.CurrentTargetFramework};netstandard2.0"));
+            propertyGroupNode.Add(new XElement("IsPublishable", "false"));
+            stream.SetLength(0);
+            await document.SaveAsync(stream, SaveOptions.None, CancellationToken.None);
+        }
+
+        // Publish
+        CommandResult commandResult = new DotnetCommand(_testOutput, "publish", "/t:PublishContainer", "-bl")
```

Comment:
why do you need `-bl` here?

Reply:
for testing locally - I can remove it, but it's not really hurting anything :)